src/Mapper/ProgrammesDbToDomain/PromotionMapper.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain;
+
+use BBC\ProgrammesPagesService\Domain\Entity\PromotableInterface;
+use BBC\ProgrammesPagesService\Domain\Entity\Promotion;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Synopses;
+use InvalidArgumentException;
+
+class PromotionMapper extends AbstractMapper
+{
+    /**
+     * @param array[] $dbPromotion
+     */
+    public function getDomainModel(array $dbPromotion): Promotion
+    {
+        $cacheKey = $this->getCacheKeyForPromotion($dbPromotion);
+        if (!isset($this->cache[$cacheKey])) {
+            $this->cache[$cacheKey] = $this->getModelForPromotion($dbPromotion);
+        }
+
+        return $this->cache[$cacheKey];
+    }
+
+    /**
+     * @param array[] $dbPromotion
+     */
+    private function getCacheKeyForPromotion(array $dbPromotion): string
+    {
+        return $this->buildCacheKey(
+            $dbPromotion,
+            'id',
+            [
+                'promotionOfCoreEntity' => 'CoreEntity',
+                'promotionOfImage' => 'Image',
+            ]
+        );
+    }
+
+    /**
+     * @param array[] $dbPromotion
+     */
+    private function getModelForPromotion(array $dbPromotion): Promotion
+    {
+        return new Promotion(
+            new Pid($dbPromotion['pid']),
+            $this->getPromotableModel($dbPromotion),
+            $dbPromotion['title'],
+            $this->getSynopses($dbPromotion),
+            $dbPromotion['uri'],
+            $dbPromotion['weighting']
+        );
+    }
+
+    /**
+     * @param array[] $dbPromotion
+     */
+    private function getSynopses(array $dbPromotion): Synopses
```

Comment:
I thought something similar in here. But I think we can make just another PR that move all this getSynopses() methods to a trait. Otherwise my PR would be noisy I think mixing different things.
It appear (repeated in these places):

PromotionMapper
CoreEntityMapper
ImageMapper
RelatedLinkMapper
SegmentEventMapper